### PR TITLE
web/satellite: Notification rework

### PR DESCRIPTION
**Pull request by NikolaiYurchenko**
What: 
Changed markup and behavior of notification on web UI, tests added and changed
Why:
New design

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/store/modules/notifications.ts [hunk 1]
```diff
@@ -12,87 +12,96 @@ export const notificationsModule = {
...
-    getters: {
-        currentNotification: (state: any): DelayedNotification => {
-            return state.notificationQueue[0] ? state.notificationQueue[0] : null;
-        },
-    },
+    }
 };
+
+
+function getNotificationById(notifications: DelayedNotification[], id: string) {
```
Comment (crawter) [inline web/satellite/src/store/modules/notifications.ts]: missing return type!

File: web/satellite/tests/unit/notifications/Notification.spec.ts [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+    const deleteSpy = sinon.spy();
+
+    beforeEach(() => {
+        actions = {
+            pauseNotification: pauseSpy,
+            resumeNotification: resumeSpy,
+            deleteNotification: deleteSpy,
+        };
+
+        store = new Vuex.Store({
```
Comment (crawter) [inline web/satellite/tests/unit/notifications/Notification.spec.ts]: don't create store before each test.

import { notificationModule } from '@/store/modules/notifications;

const store = new Vuex.Store(notificationModule);

describe('mutations', () => {...

File: web/satellite/tests/unit/notifications/Notification.spec.ts [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
 import Notification from '@/components/notifications/Notification.vue';
 import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
+import { DelayedNotification } from '@/types/DelayedNotification';
+import { makeNotificationsModule } from '@/store/modules/notifications';
+import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
+
+const localVue = createLocalVue();
+
+localVue.use(Vuex);
+const pauseSpy = sinon.spy();
```
Comment (crawter) [inline web/satellite/tests/unit/notifications/Notification.spec.ts]: you do not need this spy

just use in your tests smth like

jest.spyOn(notificationModule.actions, NOTIFICATION_ACTIONS.PAUSE)...

File: web/satellite/src/components/notifications/Notification.vue [hunk 4]
```diff
@@ -20,47 +20,60 @@
 <script lang="ts">
     import { Component, Prop, Vue } from 'vue-property-decorator';
     import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
-    import { DelayedNotification } from '../../types/DelayedNotification';
+    import { DelayedNotification } from '@/types/DelayedNotification';
```
Comment (Barterio) [inline web/satellite/src/components/notifications/Notification.vue]: imports sort order

File: web/satellite/src/components/notifications/Notification.vue [hunk 5]
```diff
@@ -20,47 +20,60 @@
...
 
     @Component({})
```
Comment (Barterio) [inline web/satellite/src/components/notifications/Notification.vue]: remove redundant braces

File: web/satellite/src/components/notifications/NotificationArea.vue [hunk 6]
```diff
@@ -17,27 +17,25 @@
         }
     })
     export default class NotificationArea extends Vue {
-        public get currentNotification(): Notification {
-            return this.$store.getters.currentNotification;
+        public get notifications(): Notification {
```
Comment (Barterio) [inline web/satellite/src/components/notifications/NotificationArea.vue]: Not clear to me. We return Notification Component?
According to store, we return []DelayedNotification, return type for this method is Notification

File: web/satellite/src/store/index.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
 
+import { makeNotificationsModule } from '@/store/modules/notifications';
```
Comment (Barterio) [inline web/satellite/src/store/index.ts]: imports order

File: web/satellite/src/store/modules/notifications.ts [hunk 8]
```diff
@@ -5,94 +5,108 @@ import { NOTIFICATION_MUTATIONS } from '../mutationConstants';
 import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
 import { DelayedNotification } from '@/types/DelayedNotification';
 import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
```
Comment (Barterio) [inline web/satellite/src/store/modules/notifications.ts]: sort order

File: web/satellite/src/store/modules/notifications.ts [hunk 9]
```diff
@@ -5,94 +5,108 @@ import { NOTIFICATION_MUTATIONS } from '../mutationConstants';
...
+            },
         },
-        // Commits muttation for adding error notification
-        [NOTIFICATION_ACTIONS.ERROR]: function ({commit}: any, message: string): void {
-            const notification = new DelayedNotification(
-                () => commit(NOTIFICATION_MUTATIONS.DELETE),
-                NOTIFICATION_TYPES.ERROR,
-                message,
-            );
+        getters: {}
```
Comment (Barterio) [inline web/satellite/src/store/modules/notifications.ts]: Do we need empty ``getters`` section?

File: web/satellite/tests/unit/notifications/Notification.spec.ts [hunk 10]
```diff
@@ -1,9 +1,27 @@
 // Copyright (C) 2019 Storj Labs, Inc.
 // See LICENSE for copying information.
 
-import { mount, shallowMount } from '@vue/test-utils';
+import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
+import Vuex from 'vuex';
```
Comment (Barterio) [inline web/satellite/tests/unit/notifications/Notification.spec.ts]: sort order

**Pull request merged**